packages/api/src/controllers/asset.ts

```diff
@@ -149,21 +193,31 @@ function parseUploadUrl(
 }
 
 app.use(
-  mung.json(function cleanWriteOnlyResponses(
+  mung.jsonAsync(async function cleanWriteOnlyResponses(
     data: WithID<Asset>[] | WithID<Asset> | { asset: WithID<Asset> },
     req
   ) {
-    if (req.user.admin) {
-      return data;
+    const ingests = await req.getIngest();
+    if (!ingests.length) {
+      throw new InternalServerError("Ingest not configured");
     }
+    const ingest = ingests[0].base;
+    const toExternalAsset = (a: WithID<Asset>) =>
+      req.user.admin
+        ? withPlaybackUrls(a, ingest)
+        : db.asset.cleanWriteOnlyResponse(withPlaybackUrls(a, ingest));
+
     if (Array.isArray(data)) {
```

Comment:
This section of the code is not clear to me, what are we doing here? In what case `data` has no `id`?